packages/capabilities/src/types.ts

```diff
@@ -439,6 +442,99 @@ export interface UploadNotFound extends Ucanto.Failure {
 
 export type UploadGetFailure = UploadNotFound | Ucanto.Failure
 
+// HTTP
+export type HTTPPut = InferInvokedCapability<typeof HTTPCaps.put>
+
+// Blob
+export type Blob = InferInvokedCapability<typeof BlobCaps.blob>
+export type BlobAdd = InferInvokedCapability<typeof BlobCaps.add>
+export type ServiceBlob = InferInvokedCapability<typeof W3sBlobCaps.blob>
+export type BlobAllocate = InferInvokedCapability<typeof W3sBlobCaps.allocate>
+export type BlobAccept = InferInvokedCapability<typeof W3sBlobCaps.accept>
+
+export type BlobMultihash = Uint8Array
+export interface BlobModel {
+  digest: BlobMultihash
+  size: number
+}
+
+// Blob add
+export interface BlobAddSuccess {
+  site: {
+    'ucan/await': ['.out.ok.site', Link]
+  }
+}
+
+export interface BlobSizeOutsideOfSupportedRange extends Ucanto.Failure {
+  name: 'BlobSizeOutsideOfSupportedRange'
+}
+
+export interface AwaitError extends Ucanto.Failure {
+  name: 'AwaitError'
+}
+
+// TODO: We need Ucanto.Failure because provideAdvanced can't handle errors without it
+export type BlobAddFailure =
+  | BlobSizeOutsideOfSupportedRange
+  | AwaitError
+  | StorageGetError
+  | Ucanto.Failure
+
+export interface BlobListItem {
+  blob: BlobModel
+  insertedAt: ISO8601Date
+}
+
+// Blob allocate
+export interface BlobAllocateSuccess {
+  size: number
+  address?: BlobAddress
+}
```

Comment:
Can we please align this on the spec https://github.com/web3-storage/specs/blob/fix/align-blob-with-impl/w3-blob.md#allocate-blob ?

It is difficult to manage what is expected if we are always diverging from there. For a second invocation with different nonce, I think not creating a presigned url for something is there makes sense, but happy to reconsider

Reply:
I already raised this a week ago https://github.com/web3-storage/w3up/pull/1342#discussion_r1553944549

I would prefer to get this merged and list what we need to improve, rather than blocking iterative approach on this huge piece of work where there is always something else and details in spec never stable

Reply:
1. The whole rename thing was just a thought, I was not implying that we should do it, sorry for not been clear about this. I just wanted to know if you it made sense to you, if so we can go update spec and then implementation as you say here.
2. You are also correct to point out that `address?: BlobAddress` comes from spec. I was honestly surprised looking at the spec now, clearly I have overlooked the issue I have then pointed out in regards to await's. I'll fix the spec and we can change this in the impl as followup.